exporter/clickhouseexporter/internal/metrics_model.go

```diff
@@ -0,0 +1,463 @@
+		ttlExpr = fmt.Sprintf(`TTL toDateTime(TimeUnix) + toIntervalDay(%d)`, ttlDays)
+	}
+	for _, table := range supportMetricsType {
+		query := fmt.Sprintf(table, tableName, ttlExpr)
+		if _, err := db.Exec(query); err != nil {
+			return fmt.Errorf("exec create gauge table sql: %w", err)
+		}
+	}
+	return nil
+}
+
+func CreateMetricsModel(tableName string) map[pmetric.MetricType]MetricsModel {
+	metricsMap := make(map[pmetric.MetricType]MetricsModel)
+	metricsMap[pmetric.MetricTypeGauge] = &GaugeMetrics{
+		InsertSQL: fmt.Sprintf(strings.ReplaceAll(insertGaugeTableSQL, "'", "`"), tableName),
+	}
+	metricsMap[pmetric.MetricTypeSum] = &SumMetrics{
+		InsertSQL: fmt.Sprintf(strings.ReplaceAll(insertSumTableSQL, "'", "`"), tableName),
+	}
+	metricsMap[pmetric.MetricTypeHistogram] = &HistogramMetrics{
+		InsertSQL: fmt.Sprintf(strings.ReplaceAll(insertHistogramTableSQL, "'", "`"), tableName),
+	}
+	metricsMap[pmetric.MetricTypeExponentialHistogram] = &ExpHistogramMetrics{
+		InsertSQL: fmt.Sprintf(strings.ReplaceAll(insertExpHistogramTableSQL, "'", "`"), tableName),
+	}
+	metricsMap[pmetric.MetricTypeSummary] = &SummaryMetrics{
+		InsertSQL: fmt.Sprintf(strings.ReplaceAll(insertSummaryTableSQL, "'", "`"), tableName),
+	}
+	return metricsMap
+}
+
+func InjectMetaData(metricsMap map[pmetric.MetricType]MetricsModel, metaData *MetricsMetaData) {
+	for _, metrics := range metricsMap {
+		metrics.InjectMetaData(metaData)
+	}
+}
+
+func InsertMetrics(ctx context.Context, tx *sql.Tx, metricsMap map[pmetric.MetricType]MetricsModel, logger *zap.Logger) error {
+	var errs []error
+	for _, metrics := range metricsMap {
+		errs = append(errs, metrics.Insert(ctx, tx, logger))
+	}
+	return multierr.Combine(errs...)
+}
+
+func convertExemplars(exemplars pmetric.ExemplarSlice) (clickhouse.ArraySet, clickhouse.ArraySet, clickhouse.ArraySet, clickhouse.ArraySet, clickhouse.ArraySet, clickhouse.ArraySet) {
+	var (
+		attrs       clickhouse.ArraySet
+		times       clickhouse.ArraySet
+		floatValues clickhouse.ArraySet
+		intValues   clickhouse.ArraySet
+		traceIDs    clickhouse.ArraySet
+		spanIDs     clickhouse.ArraySet
+	)
+	for i := 0; i < exemplars.Len(); i++ {
+		exemplar := exemplars.At(i)
+		if !exemplar.TraceID().IsEmpty() && !exemplar.SpanID().IsEmpty() {
+			attrs = append(attrs, attributesToMap(exemplar.FilteredAttributes()))
+			times = append(times, exemplar.Timestamp().AsTime())
+			floatValues = append(floatValues, exemplar.DoubleValue())
+			intValues = append(intValues, exemplar.IntValue())
+			traceID := exemplar.TraceID()
+			traceIDs = append(traceIDs, hex.EncodeToString(traceID[:]))
+			spanID := exemplar.SpanID()
+			spanIDs = append(spanIDs, hex.EncodeToString(spanID[:]))
+		}
+	}
+	return attrs, times, floatValues, intValues, traceIDs, spanIDs
+}
+
+func attributesToMap(attributes pcommon.Map) map[string]string {
+	m := make(map[string]string, attributes.Len())
+	attributes.Range(func(k string, v pcommon.Value) bool {
+		m[k] = v.Str()
```

Comment:
Not all values are strings.

Reply:
In most scenarios, attribute value is string, if it contain other types, the `Str()` will return a empy strings, I think it's ok for current usecase.

Reply:
please put a comment to that effect and add tests for this function

Reply:
done